gravitee-apim-console-webui/src/portal/customization/catalog/category-list/category-list.component.ts

```diff
@@ -0,0 +1,227 @@
+  @ViewChild('table', { static: true }) table: MatTable<Category>;
+  categoriesDS$: Observable<Category[]> = of([]);
+  private categoryList = new BehaviorSubject(1);
+  displayedColumns: string[] = ['order', 'name', 'description', 'count', 'actions'];
+
+  portalSettingsForm: FormGroup<{ enabled: FormControl<boolean> }>;
+  portalSettingsInitialValue: unknown;
+  private destroyRef = inject(DestroyRef);
+
+  constructor(
+    private categoryService: CategoryService,
+    private environmentSettingsService: EnvironmentSettingsService,
+    private readonly snackBarService: SnackBarService,
+    private matDialog: MatDialog,
+    private readonly permissionService: GioPermissionService,
+    private portalSettingsService: PortalSettingsService,
+  ) {}
+
+  ngOnInit(): void {
+    // If read-only, remove actions column
+    if (!this.permissionService.hasAnyMatching(['environment-category-u', 'environment-category-d'])) {
+      this.displayedColumns.pop();
+    }
+
+    this.initializeSettings(this.environmentSettingsService.getSnapshot().portal.apis.categoryMode.enabled);
+
+    this.categoriesDS$ = this.categoryList.pipe(
+      switchMap((_) => this.categoryService.list(true)),
+      map((categories) => categories.sort((a, b) => a.order - b.order)),
+      catchError((_) => of([])),
+    );
+  }
+
+  deleteCategory(category: Category) {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: 'Delete Category',
+          content: `Are you sure you want to delete the category '${category.name}'?`,
+          confirmButton: 'Delete',
+        },
+        role: 'alertdialog',
+        id: 'confirmDialog',
+      })
+      .afterClosed()
+      .pipe(
+        filter((confirmed) => confirmed),
+        switchMap((_) => this.categoryService.delete(category.id)),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe({
+        next: (_) => {
+          this.snackBarService.success(`'${category.name}' deleted successfully`);
+          this.categoryList.next(1);
+        },
+        error: ({ error }) => this.snackBarService.error(error?.message ?? 'Error during deletion'),
+      });
+  }
+
+  showCategory(category: Category) {
+    this.updateCategoryVisibility(false, category);
+  }
+
+  hideCategory(category: Category) {
+    this.updateCategoryVisibility(true, category);
+  }
+
+  saveSettings() {
+    this.portalSettingsService
+      .get()
+      .pipe(
+        switchMap((settings) => {
+          const newSettings = { ...settings };
+          newSettings.portal.apis.categoryMode.enabled = this.portalSettingsForm.getRawValue().enabled;
+          return this.portalSettingsService.save(newSettings);
+        }),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe(({ portal }) => {
+        this.initializeSettings(portal.apis.categoryMode.enabled);
+      });
+  }
```

Comment:
The value 'categoryMode.enabled` is pertinent to the old Portal. There shouldn't be any changes to the settings.

Reply:
Removed